newIDE/app/src/ObjectEditor/ObjectEditorDialog.js

```diff
@@ -96,8 +99,33 @@ const InnerDialog = (props: InnerDialogProps) => {
   const EditorComponent: ?React.ComponentType<EditorProps> =
     props.editorComponent;
 
-  const onApply = () => {
+  const onApply = async () => {
     props.onApply();
+
+    const changeset = gd.WholeProjectRefactorer.computeChangesetForVariablesContainer(
+      props.project,
+      getOriginalContentSerializedElement().getChild('variables'),
+      props.object.getVariables()
+    );
+    if (changeset.hasRemovedVariables()) {
+      const shouldRemoveVariables = await showConfirmation({
+        title: t`Remove actions and conditions?`,
+        message: t`You've removed some variables. Do you want to also remove all the actions and conditions in events using them?`,
+        confirmButtonLabel: t`Don't remove anything else`,
+        dismissButtonLabel: t`Delete all references to these variables`,
```

Comment:
Having long terms is nice, but this will look ugly. 
Either have shorter button labels, or find a way to have this dialog larger so the buttons don't appear on 2 lines?

Also, I kind of feel we should not ask this and always **not** remove variables